lib/thrift/transport/ssl.ex

```diff
@@ -43,26 +49,23 @@ defmodule Thrift.Transport.SSL do
       {:ok, extra_opts} ->
         {:ok, extra_opts ++ opts}
 
-      {:error, _} = error ->
+      {:error, %_exception{}} = error ->
```

Comment:
Perhaps `Exception.exception?` if we want to check this as an exception is more than `struct`.

Reply:
For now I've just matched the same pattern as in protocol_handler.ex and client.ex.